aspnetcore/fundamentals/startup.md

```diff
@@ -1,75 +1,106 @@
+The `Startup` class:
 
-Alternatively, you can define a fixed `Startup` class that will be used regardless of the environment by calling `UseStartup<TStartup>`. This is the recommended approach.
+* Must include a [Configure](/dotnet/api/microsoft.aspnetcore.hosting.startupbase.configure) method to create the app's request processing pipeline.
+* Can optionally include a [ConfigureServices](/dotnet/api/microsoft.aspnetcore.hosting.startupbase.configureservices) method to configure the app's services.
 
-The `Startup` class constructor can accept dependencies that are provided through [dependency injection](xref:fundamentals/dependency-injection). A common approach is to use `IHostingEnvironment` to set up [configuration](xref:fundamentals/configuration/index) sources.
+`Configure` and `ConfigureServices` are called when the app starts. These methods are described in more detail later in this topic.
 
-The `Startup` class must include a `Configure` method and can optionally include a `ConfigureServices` method, both of which are called when the application starts. The class can also include [environment-specific versions of these methods](xref:fundamentals/environments#startup-conventions). `ConfigureServices`, if present, is called before `Configure`.
+The app can define separate `Startup` classes for different environments (for example, `StartupDevelopment` and `StartupProduction`), and the appropriate startup class is selected at runtime. The class whose name suffix matches the current environment is prioritized. If the app is run in the Development environment and includes both a `Startup` class and a `StartupDevelopment` class, the `StartupDevelopment` class is used. See [Working with multiple environments](xref:fundamentals/environments#startup-conventions) for more information.
 
-Learn about [handling exceptions during application startup](xref:fundamentals/error-handling#startup-exception-handling).
+The `Startup` class constructor can accept dependencies that are provided through [dependency injection (DI)](xref:fundamentals/dependency-injection). A common use of DI with the `Startup` class is to inject [IHostingEnvironment](/dotnet/api/Microsoft.AspNetCore.Hosting.IHostingEnvironment) to set up [configuration sources](xref:fundamentals/configuration/index). See [Hosting: IHostingEnvironment interface](xref:fundamentals/hosting?tabs=aspnetcore2x#ihostingenvironment-interface) for an example.
+
+To learn more about `WebHostBuilder`, see the [Hosting](xref:fundamentals/hosting) topic. For information on handling errors during startup, see [Startup exception handling](xref:fundamentals/error-handling#startup-exception-handling).
```

Comment:
I'm changed the organization of this whole lead-in on the last commit. I don't think it's out of place tho to talk about the ctor before talking about the methods, and it probably doesn't warrant a separate section just to make a quick remark that one can DI stuff into the class there. I think this will look/read better after the new example is taken into account. Let's look again after the next commit.